flopy/mf6/utils/codegen/component.py

```diff
@@ -0,0 +1,57 @@
+from typing import (
+    Iterator,
+    TypedDict,
+)
+
+from modflow_devtools.dfn import Dfn
+
+
+def get_component_names(dfn: dict) -> list[tuple[str, str]]:
+    """
+    Get the names of components produced by the definition.
+    A definition may produce one or more component classes.
+    """
+    name = dfn.get("name", None)
+    if not name:
+        raise ValueError(f"DFN must have a 'name' entry")
+    name = name.split("-")
+    if name[1] == "nam":
+        if name[0] == "sim":
+            return [
+                (None, name[1]),  # nam pkg
+                tuple([*name]),  # simulation
+            ]
+        else:
+            return [
+                tuple([*name]),  # nam pkg
+                (name[0], None),  # model
+            ]
```

Comment:
Would this make it easier to read?

```suggestion
    if name == ["nam","sim"]:
        return [(None, "nam"), ("nam", "sim")
    if name[1] == "nam":
        return [
            tuple([*name]),  # nam pkg
            (name[0], None),  # model
        ]
```

Reply:
throwaway code. it can go away when we figure out how to deduplicate e.g. `mfnam.py` and `mfsimulation.py`, i.e. enforce a 1-1 mapping between dfn files and generated components.